src/IO/S3/URI.cpp

```diff
@@ -101,6 +102,19 @@ URI::URI(const std::string & uri_)
     }
     else
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Bucket or key name are invalid in S3 URI.");
+
+    if(key.empty())
+    {
+        key = "default/";
+        LOG_WARNING(&Poco::Logger::get("URI"),"The key of S3 endpoint is empty, we complete this key with 'default/' automatically. key:{}",key);
+    }
+    
+    if(key.back()!='/')
```

Comment:
Here, you need to also add a space after `if` and before and after `!=` operator:
```suggestion
    if (key.back() != '/')
```